src/context/CharactersContext.tsx

```diff
@@ -1,22 +1,62 @@
-import React, { createContext, useContext, useState, useEffect } from 'react';
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from 'react';
+import {
+  ICharacter,
+  ICharacterContext,
+  ICharacterFilter,
+} from '../types/types';
 import { useSearchParams, useNavigate } from 'react-router-dom';
 
 import { ALL_SPECIES_NAME, FILTER_NAMES } from '../constants';
 import { useCharactersApi } from '../hooks/useCharactersApi';
 import { removeEmptyKeys } from '../utils';
 
+// 1. Defining the initial state of the context
+const defaultContextValue: ICharacterContext = {
+  characters: [],
+  error: null,
+  totalCount: 0,
+  hasNextPage: false,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleNextPage: () => {},
+  isValidating: false,
+  speciesList: [],
+  selectedSpecies: [],
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleChipClick: () => {},
+  filteredCharacters: [],
+  searchParams: new URLSearchParams(),
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleApplyFilters: () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleResetFilters: () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleSearchNavigate: () => {},
+};
+
 // Create CONTEXT
-const CharactersContext = createContext();
+const CharactersContext = createContext<ICharacterContext>(defaultContextValue);
 
 // Create PROVIDER
-export const CharactersContextProvider = ({ children }) => {
+export const CharactersContextProvider = ({
```

Comment:
[req] `CharactersContextProvider` is also a component
```suggestion
export const CharactersContextProvider: React.FC<React.PropsWithChildren> = ({
```